File: csharp/Confiqure.cs
using System;

namespace Confiqure;

/// <summary>
/// Marks a class as a confiqure.ai configuration target.
///
/// <example>
/// <code>
/// [Confiqure(
///     End      = "/notifications",
///     Type     = "single",
///     Tools    = new[] { "Amazon_Connect", "Ebay_Connect" },
///     Callback = "https://myapp.com/webhooks/confiqure"
/// )]
/// public class Notifications { /* ... */ }
/// </code>
/// </example>
///
/// The confiqure CLI scans source files for <c>[Confiqure(...)]</c>. The backend
/// AI parses the annotated class and generates the chat playbook.
/// </summary>
[AttributeUsage(AttributeTargets.Class, AllowMultiple = false, Inherited = false)]
public sealed class ConfiqureAttribute : Attribute
{
    /// <summary>Chat endpoint segment. Defaults to snake_case of the class name when blank.</summary>
    public string End { get; set; } = "";

    /// <summary>Configuration shape. Defaults to "single".</summary>
    public string Type { get; set; } = "single";

    /// <summary>Names of workspace tools referenced during chat.</summary>
    public string[] Tools { get; set; } = Array.Empty<string>();

    /// <summary>End-of-chat callback URL. Defaults to the workspace's defaultCallbackUrl when blank.</summary>
    public string Callback { get; set; } = "";

    public ConfiqureAttribute() { }
}

File: kotlin/Confiqure.kt
package ai.confiqure

/**
 * Marks a class as a confiqure.ai configuration target.
 *
 * Example:
 *
 * ```
 * @Confiqure(
 *     end      = "/notifications",
 *     type     = "single",
 *     tools    = ["Amazon_Connect", "Ebay_Connect"],
 *     callback = "https://myapp.com/webhooks/confiqure"
 * )
 * class Notifications { /* ... */ }
 * ```
 *
 * The confiqure CLI scans for this annotation; the backend AI parses the
 * annotated class and generates the chat playbook.
 */
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
annotation class Confiqure(
    val end: String = "",
    val type: String = "single",
    val tools: Array<String> = [],
    val callback: String = ""
)
